### PR TITLE
Add more development-oriented documentation

### PR DESCRIPTION
**Pull request by freeekanayaka**
This branch adds a new DESIGN.txt file holding basic information about fake-juju's design and code organization. It should make it easier for people not familiar with the code (or who haven't looked at it for a while) to get a grasp on it.

It also moves some development-related documentation from README.rst to HACKING.rst, complenting it with missing bits, such else creating a new quilt patch.

### CODE REVIEW

File: DESIGN.rst [hunk 1]
```diff
@@ -0,0 +1,54 @@
+Code organization
+=================
+
+The repository keeps multiple versions of fake-juju, each building
+against the associated real-juju code.
```
Comment (albertodonato) [inline DESIGN.rst]: I'd s/associated/corresponding/

File: DESIGN.rst [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+In this way, fake-juju can be used to write integration tests that for
```
Comment (albertodonato) [inline DESIGN.rst]: s/that//

File: DESIGN.rst (hunk 1 above)
Comment (freeekanayaka) [inline DESIGN.rst]: I might be wrong, but I believe "corresponding" has a slightly different meaning in English than in Italian. I changed this with "relevant".

File: DESIGN.rst (hunk 2 above)
Comment (freeekanayaka) [inline DESIGN.rst]: Fixed.

**Pull request merged**